substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrEvent.java

```diff
@@ -101,6 +102,6 @@ public boolean shouldEmit(long durationTicks) {
 
     @Uninterruptible(reason = "Prevent races with VM operations that start/stop recording.", callerMustBe = true)
     private boolean shouldEmit0() {
-        return SubstrateJVM.get().isRecording() && SubstrateJVM.get().isEnabled(this) && !SubstrateJVM.get().isCurrentThreadExcluded();
+        return SubstrateJVM.get().isRecording() && SubstrateJVM.get().isEnabled(this) && !JfrThreadLocal.isThreadExcluded(JavaThreads.getCurrentThreadOrNull());
```

Comment:
@roberttoyonaga can you please check if this is correct?

Reply:
Yes this is correct. But https://github.com/graalvm/mandrel/pull/633/files/f3864b7fbb9d325ae604fdb0ffab0e71506bf636#diff-5140a589f5128cadb659640373aafca28c66b801fef8c975d65c08fb9eded219R105  `!SubstrateJVM.get().isCurrentThreadExcluded();` needs to be removed because now the checking is done in `JfrThreadLocal`

Reply:
oh it looks like you already removed it

Reply:
I remember there was some weird quirk that may have required `isThreadExcluded` to be checked before  `isRecording` and `isEnabled`. Let me quickly investigate to be sure.

Reply:
Ok this should be correct for 23.0.